### DISCUSSION TITLE
When combining tree sequences, how do we identify whether populations are the same?

### DISCUSSION THREAD

**Discussion by hyanwong** [tskit-dev/tskit#1915] [Q&A] [answered]
If we merge or combine two tree sequences, e.g. using `union` or if we specify an initial state to msprime, how do we know whether the population in the first file corresponds to the same population in the second file or in the simulation, or whether population 0 in ts1 is actually a different population to population 0 in ts2? I guess we could compare the metadata (or just the name) and if it's the same, assume they should both point to the same population in the output TS (and if not, create 2 separate populations)? If that's what we do, is there a build-in routine that does this somehow?

Note that the same goes for individuals, although I think nearly all the time we assume that we *aren't* referring to the same individuals in 2 tree sequences that are combined in some way. But logically it could be possible, I suppose.

**Comment by petrelharp** [comment, marked as answer]
We have no systematic way to do this - it depends on context, I think. In `union` we can check whether all the pairs of nodes in the two tree sequences that are supposed to be the "same" node point to the "same" populations; and if the metadata of populations is the same we can check that also.

**Comment by jeromekelleher** [comment]
In msprime we check that the names match up when running a simulation with a Demography and a specified initial state: https://github.com/tskit-dev/msprime/blob/b822b647dacca23b1cac598e04e49c18b0a17102/msprime/ancestry.py#L960

**Reply by hyanwong** [reply]
Ah, that's useful, thanks Jerome.
